### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Jo-Zh**
Merge to Main:
- build plugin
- dynamic data

### CODE REVIEW

File: src/ViewerPlugin.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { App } from "vue";
+import { createPinia } from "pinia";
+import CrucibleComponent from "./components/CrucibleComponent.vue";
+import { ViewerPluginOptions } from "./types";
+
+const defaultDataLink = "http://localhost:8080/api/resource/getQuiz";
```
Comment (anzau) [inline src/ViewerPlugin.ts]: Why is the address hardcoded like this? Isn't this more of an env option?

Comment (Jo-Zh) [reply src/ViewerPlugin.ts]: fixed, should not be a api link.

File: src/components/CrucibleComponent.vue [hunk 2]
```diff
@@ -1,27 +1,56 @@
...
+// Inject data from crucible parent here
+const apiData: DataApi = inject("$dataLink") as DataApi;
+
+onBeforeMount(() => {
+  // Fetch quiz data from API
+  const useStatic = import.meta.env.VITE_USE_DUMMY_DATA === "false";
+  questions.value = useStatic
+    ? getConvertedStaticData()
+    : getAllQuestions(apiData.data.questions as DataMCQuestion[]);
+  console.info("All Questions:", questions.value);
```
Comment (anzau) [inline src/components/CrucibleComponent.vue]: Why do we console.info - This is going to production, do we really need this?

File: src/components/QuestionStore.ts [hunk 3]
```diff
@@ -20,44 +20,59 @@ export const getQuestionsRandomly = (
...
-      (selectedTags.course.length === 0 ||
-        selectedTags.course.includes(question.tags.course)) &&
-      (selectedTags.subject.length === 0 ||
-        selectedTags.subject.includes(question.tags.subject)) &&
-      (selectedTags.system.length === 0 ||
-        selectedTags.system.includes(question.tags.system)) &&
-      (selectedTags.animal.length === 0 ||
-        selectedTags.animal.includes(question.tags.animal))
-    );
+  return questions.filter((question: MCQuestion) => {
```
Comment (anzau) [inline src/components/QuestionStore.ts:56]: These return statements could be simplified - Another ticket though.

File: src/components/QuestionStore.ts [hunk 4]
```diff
@@ -20,44 +20,59 @@ export const getQuestionsRandomly = (
...
   });
 }
+
+export function findSelectedOptionValue(
+  quizStats: QuestionState[],
+  questionIndex: number,
+  answer: string,
+): number | undefined {
+  const optionsList = quizStats[questionIndex].question.optionsList;
+  for (let i = 0; i < optionsList.length; i++) {
```
Comment (anzau) [inline src/components/QuestionStore.ts:72]: Should've been a `for of` for consistency but doesn't interfere functionality.

File: src/utils/NetworkGuard.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+function isMCQOptions(obj: DataMCQOptions): boolean {
+  return (
+    validate.isObject(obj) &&
+    validate.isString(obj.optionValue) &&
+    (obj.optionCorrect === undefined || validate.isBoolean(obj.optionCorrect))
+  );
+}
+
+function isMCQuestion(obj: DataMCQuestion): boolean {
+  return (
```
Comment (anzau) [inline src/utils/NetworkGuard.ts:56]: I think I mentioned this in another ticket, but this definitely needs to be simplified

**Pull request merged**